src/vec_cast.h

```diff
@@ -51,13 +51,10 @@ inline void pack_next(
     int n,
     size_t size)
 {
-    int i;
-    std::vector<Tw*> tmp(n, nullptr);
-    for (i = 0; i < n; i++) {
-        tmp[i] = reinterpret_cast<Tw*>(src.at(i));
-        std::copy_n(tmp[i], size, dest.at(i));
+    for (int i = 0; i < n; i++) {
+        Tw* tmp = reinterpret_cast<Tw*>(src.at(i));
```

Comment:
What can be the types of `Tw` and `Td` here?

(Just to be sure that we aren't breaking the strict aliasing rule)

Reply:
Then I think that we have an issue with strict aliasing violation here.

Not introduced by this PR so we can fix it later, but maybe open an issue about it so we don't forget.